src/components/ec-checkbox/ec-checkbox.spec.js

```diff
@@ -0,0 +1,116 @@
+import { mount, createLocalVue } from '@vue/test-utils';
+import EcCheckbox from './ec-checkbox.vue';
+
+function mountCheckbox(props, mountOpts) {
+  return mount(EcCheckbox, {
+    propsData: { ...props },
+    ...mountOpts,
+  });
+}
+
+function mountCheckboxAsTemplate(template, props, mountOpts) {
+  const localVue = createLocalVue();
+
+  const Component = localVue.extend({
+    components: { EcCheckbox },
+    template,
+  });
+
+  return mount(Component, {
+    localVue,
+    propsData: { ...props },
+    ...mountOpts,
+  });
+}
+
+describe('EcCheckbox', () => {
+  describe(':props', () => {
+    it(':labelMessage - should render the checkbox with a label', () => {
+      const wrapper = mountCheckbox({
+        labelMessage: 'Test label message prop',
+      });
+
+      expect(wrapper.find('.ec-checkbox__label').exists()).toBe(true);
+      expect(wrapper.element).toMatchSnapshot();
+    });
+
+    it(':errorMessage - should render the checkbox with an errorMessage', () => {
+      const wrapper = mountCheckbox({
+        errorMessage: 'Test error message prop',
+      });
+
+      expect(wrapper.find('.ec-checkbox__error-text').exists()).toBe(true);
+      expect(wrapper.element).toMatchSnapshot();
+    });
+
+    it(':large - should render a large checkbox', () => {
+      const wrapper = mountCheckbox({
+        large: true,
+      });
+
+      expect(wrapper.find('.ec-checkbox__label--large').exists()).toBe(true);
+      expect(wrapper.element).toMatchSnapshot();
+    });
+  });
+
+  describe('#slots', () => {
```

Comment:
Maybe we should verify race conditions when I pass error-message slot and error-message prop together, which one I should see? the slot.

Reply:
Is similar to what I answered Carlos above. Should we hide one of the 2 and make it "dummy-proof". IMHO I don't like this. The person that writing the "HTML" should get a visual clue that he is doing something wrong.